atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -106,21 +108,32 @@ def __init__(self, tasker, workflow, kojihub, url,
         self.kojihub = kojihub
         self.koji_ssl_certs = koji_ssl_certs
         self.koji_proxy_user = koji_proxy_user
-
         self.koji_principal = koji_principal
         self.koji_keytab = koji_keytab
 
+        self.use_auth = use_auth
+        self.url = url
+        self.verify_ssl = verify_ssl
+
         self.blocksize = blocksize
         self.target = target
         self.poll_interval = poll_interval
 
         self.namespace = get_build_json().get('metadata', {}).get('namespace', None)
-        osbs_conf = Configuration(conf_file=None, openshift_uri=url,
-                                  use_auth=use_auth, verify_ssl=verify_ssl,
-                                  namespace=self.namespace)
-        self.osbs = OSBS(osbs_conf, osbs_conf)
+        self.osbs = self._get_openshift_session()
         self.build_id = None
 
+    def _get_openshift_session(self):
+        try:
+            return get_openshift_session(self.workflow, self.namespace)
+        except KeyError:
+            pass
+
+        osbs_conf = Configuration(conf_file=None, openshift_url=self.url,
+                                  use_auth=self.use_auth, verify_ssl=self.verify_ssl,
+                                  namespace=self.namespace)
+        return OSBS(osbs_conf, osbs_conf)
+
```

Comment:
This code is repeated in several other plugins. It should be refactored and moved into util.py